### DISCUSSION TITLE
Show all variables for a build

### DISCUSSION THREAD

**Discussion by wijzijnweb** [codemagic-ci-cd/codemagic-docs#2523] [Ideas]
### Use case

```Text
When I submit a new build through the API and add some environment variables to it I can't see them in codemagic. Making me wonder if they are set correctly, until a build fails or not.
```

### Idea

```Text
Some button after which we can see the variables that have been set.
```

**Comment by helinanever** [comment]
Thank you for this feature request, we can see this being useful and have added it to our backlog. We will keep you posted on any updates.

**Comment by helinanever** [comment]
Hi @wijzijnweb, we're now logging user-defined variables as the first item during the Preparing build machine step. Secret values are hidden, showing a hash instead.

I'm going to close this request as completed. Let us know if you have any feedback.
